pkg/apis/aws/validation/worker.go

```diff
@@ -120,3 +122,29 @@ func validateVolumeConfig(volume *apisaws.Volume, volumeType string, fldPath *fi
 
 	return allErrs
 }
+
+func validateInstanceMetadata(md *apisaws.InstanceMetadata, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if md == nil {
+		return allErrs
+	}
+
+	if !md.EnableInstanceMetadataV2 && md.HTTPPutResponseHopLimit != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("HTTPPutResponseHopLimit"), *md.HTTPPutResponseHopLimit, "enableInstanceMetadataV2 must be set to specify this field"))
+		return allErrs
+	}
+
+	if md.EnableInstanceMetadataV2 && md.HTTPPutResponseHopLimit == nil {
```

Comment:
Can this case ever happen as `HTTPPutResponseHopLimit` is defaulted to 2?

Reply:
Correct it cannot happen as it is (due to the defaulter). This is a bit of "defensive programming".